File: backend/src/main/java/com/eskisehir/eventapi/dto/RefreshTokenRequest.java
package com.eskisehir.eventapi.dto;

import jakarta.validation.constraints.NotBlank;

/**
 * DTO for refresh token requests.
 * Used to exchange a refresh token for a new access token.
 */
public class RefreshTokenRequest {

    @NotBlank(message = "Refresh token is required")
    private String refreshToken;

    public RefreshTokenRequest() {
    }

    public RefreshTokenRequest(String refreshToken) {
        this.refreshToken = refreshToken;
    }

    public String getRefreshToken() {
        return refreshToken;
    }

    public void setRefreshToken(String refreshToken) {
        this.refreshToken = refreshToken;
    }
}

File: mobile/app/src/main/java/com/eskisehir/eventapp/data/model/RefreshTokenRequest.kt
package com.eskisehir.eventapp.data.model

import com.google.gson.annotations.SerializedName

/**
 * DTO for refresh token requests.
 * Used to exchange a refresh token for a new access token.
 */
data class RefreshTokenRequest(
    @SerializedName("refreshToken")
    val refreshToken: String
)
